### PR TITLE
refactor(Summary): refactor Summary and ElementHelper to group AddAnother fieldsets

### PR DESCRIPTION
**Pull request by lydiagray**
### Description
- refactor GenerateQuestionAndAnswersList to generate keys for AddAnother fieldsets
- refactor Summary to insert sections for AddAnother fieldsets

### Checklist
- [x] Code compiles correctly
- [x] Created tests for the new changes
- [x] All tests passing
- [x] Extended the README / documentation, if necessary

### CODE REVIEW

File: src/Helpers/ElementHelpers/ElementHelper.cs [hunk 1]
```diff
@@ -203,49 +203,71 @@ public List<PageSummary> GenerateQuestionAndAnswersList(string guid, FormSchema
...
                 formSummary.Add(pageSummary);
             }
 
             return formSummary;
         }
 
-        public Dictionary<string, string> GenerateSummaryAnswers(List<IElement> formSchemaQuestions, Page page, FormAnswers formAnswers)
+        public int GetAddAnotherNumberOfFieldsets(IElement addAnotherElement, FormAnswers formAnswers)
+        {
+            var formDataIncrementKey = $"addAnotherFieldset-{addAnotherElement.Properties.QuestionId}";
```
Comment (lallysmbc) [inline src/Helpers/ElementHelpers/ElementHelper.cs:251]: move addAnotherFieldset to constant

File: src/Helpers/ElementHelpers/ElementHelper.cs [hunk 2]
```diff
@@ -203,49 +203,71 @@ public List<PageSummary> GenerateQuestionAndAnswersList(string guid, FormSchema
...
+            return formAnswers.FormData.ContainsKey(formDataIncrementKey) 
+                ? int.Parse(formAnswers.FormData.GetValueOrDefault(formDataIncrementKey).ToString()) 
+                : throw new ApplicationException($"ElementHelper::GetCurrentAddAnotherIncrement, FormData key not found for {formDataIncrementKey}");
```
Comment (lallysmbc) [inline src/Helpers/ElementHelpers/ElementHelper.cs:254]: would this throw if only a single item is added and you proceeded straight away?

File: src/Helpers/ElementHelpers/ElementHelper.cs [hunk 3]
```diff
@@ -203,49 +203,71 @@ public List<PageSummary> GenerateQuestionAndAnswersList(string guid, FormSchema
...
+                {
+                    var listOfPageSummary = new List<PageSummary>();
+                    var addAnotherElement = page.Elements.FirstOrDefault(_ => _.Type == EElementType.AddAnother);
+                    var currentIncrement = GetAddAnotherNumberOfFieldsets(addAnotherElement, formAnswers);
+                    for (var i = 1; i <= currentIncrement; i++)
+                    {
+                        var addAnotherPageSummary = new PageSummary
+                        {
+                            PageTitle = addAnotherElement.GetLabelText(page.Title),
+                            PageSlug = $"{page.PageSlug}-{addAnotherElement.Properties.QuestionId}-{i}"
```
Comment (lallysmbc) [inline src/Helpers/ElementHelpers/ElementHelper.cs:218]: the change link should just be pageSlug, assuming the change url would not work as it has the questionID within?

File: src/Helpers/ElementHelpers/ElementHelper.cs (hunk 2 above)
Comment (lydiagray) [reply src/Helpers/ElementHelpers/ElementHelper.cs:254]: It shouldn't because the FormData gets added in ProcessAddAnother on submit

**Review by ColinLees — APPROVED** [review]

**Pull request merged**